explorer/ast/member.h

```diff
@@ -25,27 +27,112 @@ struct NamedValue {
   Nonnull<const Value*> value;
 };
 
-// A member of a type.
+// A generic member of a type.
+//
+// This is can be a named, positional or other type of member.
```

Comment:
From a terminology perspective, I don't think base classes and tuple elements are really 'members'. Can we find a better word?

'Subobject' seems good for most of the cases here, but this is also used to name methods, which aren't subobjects. Maybe 'element':

- `Element`
    - `MemberElement`
    - `TupleElement`
    - `BaseElement`

... seems OK. What do you think?

Alternatively: `FieldPath::Component` is a very similar thing. Maybe 'component' would be a good choice of name?

Whatever name we choose, I think it'd make sense to rename `FieldPath` to match (`ElementPath` or `ComponentPath` or similar).

Reply:
Component and Suboject seem a bit generic and a bit too specific respectively to me, but I like like "Element", that would be a good naming option and conveys the idea that is a piece of a whole. Good recommendation.

Reply:
Renamed with `Element`

Reply:
For what it's worth, my intuition is that tuple elements are members, and the base class subobject seems close enough to being a member that I don't find `BaseMember` problematic. Conversely, I think of "elements" as being things that are part of the object layout, so I find it weird and unfamiliar to refer to things like methods as "elements". However, I'm not sure we'll be able to find a term that satisfies both of us, and I think I can get used to "element".

However, given the uncertainty around "member" and "element", `MemberElement` seems like a pretty unhelpful name. Also, it seems kind of awkward that a `MemberElement` is an element that is a member, and a `BaseElement` is an element that is a base class subobject, but a `TupleElement` is not an element that is a tuple. Also, it seems quite likely that we'll want to be able to reuse `TupleElement` in arrays, where that name will be even more confusing.

So even if we stick with `Element` as the base class name, I think we should use `Named` and `Positional` rather than `Member` and `Tuple` in the subclass names.

Reply:
Any thoughts @zygoloid?

Reply:
`NamedElement`, `PositionalElement`, `BaseElement` seems like a reasonable approach to me, and avoids the question of whether the elements of a tuple are members or not.